### DISCUSSION TITLE
QBtnDropdown / QMenu / max-height

### DISCUSSION THREAD

**Discussion by dev31337** [quasarframework/quasar#15862] [Ideas / Proposals]
It seems that there is now way to specify the max-height property of the QMenu of a QBtnDropdown.

It is not possible to use a style="max-height: ..." attribute because the QMenu would override it (even with !important).

Would it be possible to add a max-height property to the QBtnDropdown (so this property could be forwarded to the QMenu) ?

**Comment by pdanpdan** [comment]
You render the content of the menu in the slot. Define a max-height

**Comment by dev31337** [comment]
Thank you for your reply, can you please elaborate just a bit, what do you mean by "render the content of the menu in the slot" ?

Do you mean putting my own QMenu in the "default" slot of the QBtnDropdown ?

If you have any example (even with another component) don't hesitate to give me a link to it :)

**Comment by pdanpdan** [comment]
All examples in the docs
The content of the slot is what is rendered in the menu. Set max-height on the list

**Comment by dev31337** [comment]
I read and tested everything possible. (I'm developing on Quasar since 3 years)

What is not working:

1. Putting a QMenu inside QBtnDropdown. (the Menu does not show)
2. Using a "template v-slot:default" to put the QMenu. (the Menu does not show)
3. Only putting a QList/QItem inside the QBtnDropdown as expected. (The Menu shows but there is no way to set the max-height property of the QMenu, setting a max-height style on QList or QItem does not work)

My only solution so far is to use a QButton and a QMenu inside and use the max-height property of the QMenu.

It is important to know that the QMenu overrides the max-height property depending on the height of the screen, unless there is a max-height property set in which case it does not change it.

If you tell me more about your idea I will make a codepen to show you that it does not work. (unless it does)

**Comment by dev31337** [comment]
I would like to explain what I'm doing first, I want the QMenu to display the full menu without a scrollbar, the only way to achieve it is to set the max-height property of the QMenu, any other way to set the max-height using any QList or QItem property or using style="max-height= ... !important;" on Qlist or QItem is ignored because QMenu max-height property internal management will override it.

**Comment by pdanpdan** [comment]
https://codepen.io/pdanpdan/pen/KKGGYNP?editors=1010

**Comment by dev31337** [comment]
You closed it too fast here your code pen updated which shows the issue:

https://codepen.io/Zikey/pen/LYggoPB?editors=1010

You will see the the first dropdown button will have its QMenu not going to the bottom of the screen, while the QMenu of the regular button will go to the bottom of the screen.

I dare you to make your dropdown button QMenu behave like the regular button QMenu (go the end of the screen), you will see that you can not.

**Reply by pdanpdan** [reply]
https://codepen.io/pdanpdan/pen/RweemRz

So what you wanted was to increase the maximum max-height of the menu - you cannot do it directly with `content-style` because QMenu removes it, but you can set it with a class

**Comment by dev31337** [comment]
oh nice trick, thanks a lot !

Don't you think that a max-height (or content-max-height) property on the QDropDownButton would simplify it ? :)

**Comment by pdanpdan** [comment]
Not really. The direct max-height cannot increase menu height over the default max height. You should still need the trick
